RecoLocalTracker/Phase2TrackerRecHits/plugins/VectorHitBuilderEDProducer.cc

```diff
@@ -0,0 +1,134 @@
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "RecoLocalTracker/Phase2TrackerRecHits/interface/VectorHitBuilderAlgorithm.h"
+#include "DataFormats/Phase2TrackerCluster/interface/Phase2TrackerCluster1D.h"
+#include "DataFormats/TrackerRecHit2D/interface/VectorHit.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+class VectorHitBuilderEDProducer : public edm::stream::EDProducer<>
+{
+
+ public:
+
+  explicit VectorHitBuilderEDProducer(const edm::ParameterSet&);
+  virtual ~VectorHitBuilderEDProducer();
+  virtual void produce(edm::Event&, const edm::EventSetup&) override;
+  void setupAlgorithm(edm::ParameterSet const& conf);
+  void run(edm::Handle< edmNew::DetSetVector<Phase2TrackerCluster1D> > clusters,
+           edmNew::DetSetVector<Phase2TrackerCluster1D>& clustersAcc, edmNew::DetSetVector<Phase2TrackerCluster1D>& clustersRej,
+           VectorHitCollectionNew& outputAcc, VectorHitCollectionNew& outputRej);
+//  VectorHitBuilderAlgorithm * algo() const { return builderAlgo; };
+
+ private:
+  VectorHitBuilderAlgorithm* builderAlgo;
```

Comment:
Use `unique_ptr`.

Reply:
Or even better, make it a normal member and remove `algoTag` as there are no other algorithms at the moment. Adding support for multiple algorithms in the future is relatively straightforward (and needs to be done differently if we want to support `fillDescriptions()`).